src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/ComponentVerificationTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.Composition.Reflection;
+
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    public partial class ComponentVerificationTests
+    {
+        [Theory]
+        [ClassData(typeof(ComposedPartTestData))]
+        public void ImportsMustFilterBasedOnCapabilities(Type type)
+        {   // Imports should respect/filter the capabilities of the exports they receive
+
+            var part = ComponentComposition.Instance.FindComposedPart(type);
+
+            foreach (KeyValuePair<ImportDefinitionBinding, IReadOnlyList<ExportDefinitionBinding>> binding in part.SatisfyingExports)
```

Comment:
Seems like a nice candidate for destructuring.

Reply:
👍